app/src/main/java/org/oppia/app/utility/SplitScreenManager.kt

```diff
@@ -0,0 +1,71 @@
+package org.oppia.app.utility
+
+import android.content.Context
+import android.content.res.Resources
+import org.oppia.app.R
+import org.oppia.app.model.EphemeralQuestion
+import org.oppia.app.model.EphemeralState
+import org.oppia.app.player.state.StateFragment
+import org.oppia.app.topic.questionplayer.QuestionPlayerFragment
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+private const val MINIMUM_DIAGONAL_WIDTH = 7.0
+
+/**
+ * A helper class that is used to detect whether to split the screen in [QuestionPlayerFragment]
+ * and [StateFragment] or not based on multiple factors.
+ */
+@Singleton
+class SplitScreenManager @Inject constructor(private val context: Context) {
+  private val splitScreenInteractionIdsPool = listOf("DragAndDropSortInput", "ImageClickInput")
```

Comment:
Can we centrally define this property, instead? E.g. either as part of the service-level interaction module, or as part of the interaction view model module. We have two modules that define all things interactions, and we should incorporate this in one of those locations to make it easier to add & tweak interaction support in the future.

I suggest adding this to the UI module for defining interaction view models, if possible: https://github.com/oppia/oppia-android/blob/develop/app/src/main/java/org/oppia/app/player/state/itemviewmodel/InteractionViewModelModule.kt.

Reply:
I have moved it to the companion object of the suggested class `InteractionViewModelModule`

Reply:
Resolving this on behalf of Ben.